app/src/main/java/android/example/courses/ScheduleFragment.kt

```diff
@@ -0,0 +1,175 @@
+package android.example.courses
+
+import android.content.Context
+import android.content.res.Resources
+import android.graphics.Color
+import android.os.Bundle
+import android.view.Gravity
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.LinearLayout
+import android.widget.TextView
+
+class ScheduleFragment : Fragment() {
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val container = inflater.inflate(R.layout.fragment_schedule, container, false)
+        val globalLinearLayout = container.findViewById<LinearLayout>(R.id.globalScheduleFrameLayout)
+        buildPage(globalLinearLayout)
+        return container
+    }
+
+    private fun buildPage(container: LinearLayout) {
+        val day = "Monday"
+        val totalHours = "5.34 hours"
+
+        val layout = LinearLayout(context)
+
+        val params = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.WRAP_CONTENT
+        )
+        params.setMargins(0, 10.dpToPx, 0, 0)
+
+        layout.orientation = LinearLayout.VERTICAL
+        layout.layoutParams = params
+
+        layout.addView(getDayHeader(day, totalHours))
```

Comment:
Please use RecyclerView for displaying lists instead of programmatically adding them. It's non-optimal for memory usage (and also hard to maintain, especially when data becomes dynamic and is not hardcoded).

You'll most likely need to implement a RecyclerView.Adapter that supports multiple view types. Check this out as an example: https://medium.com/@ivancse.58/android-and-kotlin-recyclerview-with-multiple-view-types-65285a254393

Reply:
@radumarin88 Thanks for this link! It's really interesting! The main purpose of schedule fragment was to experiment programmatic view management by simulating a scenario where even if the view is a list, it is too "complex" to be represented in a flat way (which is the way I thought a recycler view is appropriate for)

Basically, in this fragment I need to represent a list of lists (each day in the schedule has a variable number of schedule entries), so the only thing in my mind was to use "a recycler view of recycler views" which was definitely not a right path.

An adapter with multiple view types would work here, I guess 🤔